### DISCUSSION TITLE
nix language syntax

### DISCUSSION THREAD

**Discussion by xbz-24** [helix-editor/helix#9263] [Troubleshooting] [answered]
is it possible to add nix expressions syntax highlighting? currently when I open my nixos configuration

sudo hx /etc/nixos/configuration.nix

I see like if its a .txt, when I open it with vim/neovim I get syntax highlight but with helix not. Is the nix language supported for this editor?

**Comment by TornaxO7** [comment]
May I ask what your output is from `hx --version`? I have syntax highlighting for my `nix` files and I'm using `helix` from the master branch.

![image](https://github.com/helix-editor/helix/assets/50843046/b2023554-4df3-4d46-b988-1081957530db)

**Reply by xbz-24** [reply]
<img width="648" alt="image" src="https://github.com/helix-editor/helix/assets/68678258/4607f2ed-d11a-475f-bf32-0a84c06cb1db">

then its a me problem, may I ask how did you set it up for it to get the syntax highlighting?

**Reply by TornaxO7** [reply]
I did nothing. Hm... could you try out `hx --health nix` and post your output here?

**Reply by xbz-24** [reply]
<img width="514" alt="image" src="https://github.com/helix-editor/helix/assets/68678258/72760ba2-8c4e-453b-aeb2-1990ed48bf3e">

**Reply by TornaxO7** [reply]
hm... weird. That's my output:
![image](https://github.com/helix-editor/helix/assets/50843046/c8a86bae-03e4-4157-a07e-0667b9d194d8)

Is `nix` detected (try with `:set-language`)? If not: Have you tried to set the filetype manually? Maybe trying out another colorscheme works?

**Reply by kirawi** [reply]
Can you also check `:log-open`

**Reply by xbz-24** [reply]
![image](https://github.com/helix-editor/helix/assets/68678258/39c93798-c70a-4718-97f6-156f2f2423e1)

**Reply by kirawi** [reply]
Just to check, do you see anything when running `:tree-sitter-scopes` on some of the text?

**Reply by xbz-24** [reply]
![image](https://github.com/helix-editor/helix/assets/68678258/9dc65aaa-701f-4e0a-ad98-7c5cbb20bbf5)
I get this wehn doing ":tree-sitter-scopes" in a .nix

**Reply by gabydd** [reply]
That file looks like it has highlighting, maybe you want to use another theme that highlights more things if you want it to look more highlighted. You can look through the built-in ones with ``:theme `` and then tab through the list.

**Reply by xbz-24** [reply, marked as answer]
![image](https://github.com/helix-editor/helix/assets/68678258/11d23bee-1134-45e9-aea0-e81e5f0c0841)

you are right, I feel stupid now, thx for the help everyone

**Comment by xbz-24** [comment]
when I do :set-language nix nothing happens

![image](https://github.com/helix-editor/helix/assets/68678258/7ce2eda3-35d5-48d6-b350-a4adb91ee993)

**Reply by Brisingr05** [reply]
Have you installed nil?

**Reply by xbz-24** [reply]
> Have you installed nil?

no

**Reply by Brisingr05** [reply]
~After installing nil (e.g. by adding it to `environment.systemPackages`) syntax highlighting should start working.~ See below.

**Reply by the-mikedavis** [reply]
Installing a language server has no effect on syntax highlighting: we use tree-sitter rather than LSP for highlights

**Reply by Brisingr05** [reply]
Thanks for the correction! It seems I got confused between the two.
